server/raft.go

```diff
@@ -2212,12 +2212,13 @@ var aePool = sync.Pool{
 // appendEntry is the main struct that is used to sync raft peers.
 type appendEntry struct {
 	leader  string   // The leader that this append entry came from.
-	term    uint64   // The current term, as the leader understands it.
-	commit  uint64   // The commit index, as the leader understands it.
+	term    uint64   // The term when this entry was stored.
+	commit  uint64   // The commit index of the leader when this append entry was sent.
 	pterm   uint64   // The previous term, for checking consistency.
 	pindex  uint64   // The previous commit index, for checking consistency.
 	entries []*Entry // Entries to process.
 	// Below fields are for internal use only:
+	lterm uint64        // The highest term for catchups only, as the leader understands it. (If lterm=0, use term instead)
```

Comment:
Do we need this for every AE? Adds in 8 bytes on the wire.

Reply:
Is this only for cathcups?

Reply:
Correct, `lterm!=0` only when catching up, during normal operation it's always `lterm=0`. It will always be encoded into the append entry, but it's a `uvarint`, so it will only take up 1 byte in that case.

Reply:
We _could_ omit a zero-valued `lterm` from the wire altogether, but it makes it very hard for us to add later fields in the future if we needed to. For the sake of one byte it feels safer to always include it.